src/jarabe/desktop/activitychooser.py

```diff
@@ -0,0 +1,232 @@
+
+from jarabe.desktop.activitieslist import ActivitiesList
+from jarabe.model import bundleregistry, shell
+from jarabe.util.normalize import normalize_string
+
+_AUTOSEARCH_TIMEOUT = 1000
+
+class ActivityChooser(PopWindow):
+
+    __gtype_name__ = 'ActivityChooser'
+
+    __gsignals__ = {
+        'response': (GObject.SignalFlags.RUN_FIRST, None, ([int])),
+        'activity-selected': (GObject.SignalFlags.RUN_FIRST, None, ([object, object])),
+    }
+
+    def __init__(self):
+        logging.debug('In the Object Chooser class init hehehe')
+        PopWindow.__init__(self)
+        width, height = self.HALF_WIDTH
+
+        self.set_size((width*3/2, height*2/3))
+        self.connect('key-press-event', self.__key_press_event_cb)
+        self._list_view = ActivitiesList()
+
+        self.search_bar = SearchBar()
+        self.get_vbox().pack_start(self.search_bar, False, False    , 0)
+        self.search_bar.connect('query-changed', self.__toolbar_query_changed_cb)
+        self.search_bar.search_entry.connect('key-press-event',
+                                             self.__key_press_event_cb)
+        self._scrolled_window = Gtk.ScrolledWindow()
+        self._scrolled_window.set_policy(Gtk.PolicyType.AUTOMATIC,
+                                         Gtk.PolicyType.AUTOMATIC)
+
+        self._scrolled_window.add(self._list_view)
+
+        self.get_vbox().pack_start(self._scrolled_window, True, True, 0)
+
+        self._list_view.show()
+        self._list_view.connect('clear-clicked',
+                                self.__activitylist_clear_clicked_cb, self.search_bar)
+
+
+        self.tree_view = self._list_view._tree_view
+
+        self.tree_view.date_column.set_visible(False)
+        self.tree_view.fav_column.set_visible(False)
+        self.tree_view.version_column.set_visible(False)
+
+        if self.tree_view.row_activated_handler:
+            self.tree_view.disconnect(self.tree_view.row_activated_handler)
+        if self.tree_view.button_press_handler:
+            self.tree_view.disconnect(self.tree_view.button_press_handler)
+        if self.tree_view.button_reslease_handler:
+            self.tree_view.disconnect(self.tree_view.button_reslease_handler)
+        if self.tree_view.icon_clicked_handler:
+            self.tree_view.disconnect(self.tree_view.icon_clicked_handler)
+
+        self.tree_view.connect('row-activated',self.__row_activated_cb)
+
+        self.show()
+
+    def __toolbar_query_changed_cb(self, toolbar, query):
+        self._query = normalize_string(query.decode('utf-8'))
+        self._list_view.set_filter(self._query)
+
+        toolbar.search_entry._icon_selected = \
+            self._list_view.get_activities_selected()
+
+        # verify if one off the selected names is a perfect match
+        # this is needed by th case of activities with names contained
+        # in other activities like 'Paint' and 'MusicPainter'
+        for activity in self._list_view.get_activities_selected():
```

Comment:
I feel like this code is copped from the home view.  That's good from a UX consistency standpoint.  But can you re-use the code without copy and paste?